dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp

```diff
@@ -56,35 +53,60 @@ namespace
         clone->children.push_back(ast_func);
         return clone;
     }
+
+    ASTPtr createFakeFixedStringAst()
+    {
+        ASTPtr result = std::make_shared<ASTLiteral>(Field(UInt64(1)));
+        auto clone = result->clone();
+        clone->children.clear();
+        clone->children.push_back(result);
+        return clone;
+    }
 }
 
 void StorageSystemDataTypeFamilies::fillData(MutableColumns & res_columns) const
 {
     const auto & factory = DataTypeFactory::instance();
-    const auto & data_types = factory.getAllDataTypes();
-    for (const auto & pair : data_types)
+    auto names = factory.getAllDataTypeNames();
+    for (const auto & name : names)
     {
-        res_columns[0]->insert(pair.first);
+        res_columns[0]->insert(name);
+        res_columns[1]->insert(UInt64(factory.isCaseInsensitive(name)));
+
+        if (factory.isAlias(name))
+            res_columns[2]->insert(factory.aliasTo(name));
+        else
+            res_columns[2]->insert(String(""));
 
         try
         {
             DataTypePtr type_ptr;
-            //special case with enum, because it has arguments but it's properties doesn't
-            //depend on arguments
-            if (boost::starts_with(pair.first, "Enum"))
+
+            Field size = Null();
+            // hardcoded cases for simple parametric types
+            if (boost::starts_with(name, "Enum"))
+            {
+                type_ptr = factory.get(name, createFakeEnumCreationAst());
+                size = type_ptr->getMaximumSizeOfValueInMemory();
+            }
+            else if (name == "FixedString" || name == "BINARY")
             {
-                type_ptr = factory.get(pair.first, createFakeEnumCreationAst());
+                type_ptr = factory.get(name, createFakeFixedStringAst());
             }
             else
             {
-                type_ptr = factory.get(pair.first);
+                type_ptr = factory.get(name);
+                if (type_ptr->haveMaximumSizeOfValue())
+                    size = type_ptr->getMaximumSizeOfValueInMemory();
             }
 
-            res_columns[1]->insert(getPropertiesAsString(type_ptr));
+            res_columns[3]->insert(size);
+
+            setTypePropertries(type_ptr, res_columns);
         }
         catch (Exception & ex)
         {
-            res_columns[1]->insert(String{"depends_on_arguments"});
+            setComplexTypeProperties(name, res_columns);
```

Comment:
It's wrong to use exceptions for non-exceptional control flow.

Reply:
I have thought about this. But how can we know, that the type is complex/have nested types if we would not try to instantiate it? I see only one solution -- hardcoded types names, but I think current logic is better.